src/components/BookInfo.tsx

```diff
@@ -0,0 +1,47 @@
+import { Link, useLocation } from 'react-router-dom'
+import './BookInfo.css'
+const BookInfo = () => {
+  const location = useLocation()
+  const info: any = location.state
+  console.log(info)
```

Comment:
DO NOT add any side affects to the rendering logic or a React component. And remember console statement are for DEBUGGING and DEVELOPMENT only. They should not be committed!